pkgs/development/python-modules/openai-triton/default.nix

```
@@ -0,0 +1,253 @@
+{ lib
+, buildPythonPackage
+, python
+, fetchpatch
+, fetchFromGitHub
+, addOpenGLRunpath
+, cmake
+, cudaPackages ? { }
```

I don't think that will work since the attrs are still accessed below.

```suggestion
, cudaPackages
```

---

These attrs are only evaluated when `cudaSupport` is `true`

---

cudePackages.backendStdenv is accessed in preConfigure unconditional. 

Also cudaPackages exists in all-packages, so it always exists here.

---

> cudePackages.backendStdenv is accessed in preConfigure unconditional.

Ah, I forgot, I make it optionalString in the follow-up PR.

> Also cudaPackages exists in all-packages, so it always exists here.

Hmm, I think it was originally Fridh's idea that `cudaPackages` may not exist on some platforms, and this went into the manual. I wasn't sure I know everything about how the package set is put together, so I just eventually picked up this tradition. I'll be happy to drop it